services/watson-extension/src/watson_extension/clients/insights/rhsm.py

```diff
@@ -82,3 +90,21 @@ async def check_subscriptions(
                     )
 
         return subscriptions_info
+
+    async def create_activation_key(self, name: str):
+        body = {"name": name, "role": "", "serviceLevel": "", "usage": ""}
+        response = await self.platform_request.post(
+            self.rhsm_url,
+            "/api/rhsm/v2/activation_keys",
+            user_identity=await self.user_identity_provider.get_user_identity(),
+            json=body,
+        )
+
+        response_json = await response.json()
+
+        if not response.ok:
+            return ActivationKeyResponse(
+                ok=response.ok,
+                message=response_json.get("error", {}).get("message", "Unknown error"),
```

Comment:
We need to figure out how are we going to report to the user errors from the API - but i think it is fine for now to just raise an error with `response.raise_for_status()` 
We will probably catch the error somewehere else and say something like "We couldn't process, please try again" or similar.

If we can manage the error in the client, and do something else than the plain "try again", by all means do it, but if not, lets handle it globally.

Reply:
I tried it with response.raise_for_status(), but since an error response is ok (as in the template reads in that response), I had to remove it so it would render :thinking:

Reply:
Just to add, if you think that above template (please go here and use this error message) is going to be helpful, lets use it.

Reply:
> I tried it with response.raise_for_status(), but since an error response is ok (as in the template reads in that response), I had to remove it so it would render 🤔

If an error response is expected and we know how to manage it, then what you have seems fine!